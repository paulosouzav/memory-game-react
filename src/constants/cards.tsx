import Asteroid from '../assets/cards/002-asteroid.png';
import Earth from '../assets/cards/006-planet-earth.png';
import Mars from '../assets/cards/017-mars.png';
import Sun from '../assets/cards/033-sun.png';
import Saturn from '../assets/cards/028-saturn.png';
import Spaceship from '../assets/cards/014-space-ship.png';
import DefaultImage from '../assets/cards/default-card.png';
import { ICard } from '../components/Card';
import { IImage } from '../components/Image';

export const DEFAULT_CARD_IMAGE: IImage = {
  src: DefaultImage,
  alt: 'Default',
};

export const CARDS: Array<ICard> = [
  {
    id: 'asteroid',
    image: {
      src: Asteroid,
      alt: 'Asteroid',
    },
  },
  {
    id: 'earth',
    image: {
      src: Earth,
      alt: 'Earth',
    },
  },
  {
    id: 'mars',
    image: {
      src: Mars,
      alt: 'Mars',
    },
  },
  {
    id: 'saturn',
    image: {
      src: Saturn,
      alt: 'Saturn',
    },
  },
  {
    id: 'spaceship',
    image: {
      src: Spaceship,
      alt: 'Spaceship',
    },
  },
  {
    id: 'sun',
    image: {
      src: Sun,
      alt: 'Sun',
    },
  },
];
