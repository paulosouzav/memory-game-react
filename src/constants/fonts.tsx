import { css } from 'styled-components';
import Roboto from '../assets/fonts/Roboto-Regular.ttf';
import RobotoThin from '../assets/fonts/Roboto-Thin.ttf';
import RobotoLight from '../assets/fonts/Roboto-Light.ttf';
import RobotoBold from '../assets/fonts/Roboto-Bold.ttf';

export const FONTS = Object.freeze({
  REGULAR: 'Roboto',
  THIN: 'RobotoThin',
  LIGHT: 'RobotoLight',
  BOLD: 'RobotoBold',
});

interface FontFace {
  name: string;
  url: any;
  weight: string | number;
}

const FONT_FACES_LIST: Array<FontFace> = [
  {
    name: FONTS.REGULAR,
    url: Roboto,
    weight: 'normal',
  },
  {
    name: FONTS.THIN,
    url: RobotoThin,
    weight: 100,
  },
  {
    name: FONTS.LIGHT,
    url: RobotoLight,
    weight: 300,
  },
  {
    name: FONTS.BOLD,
    url: RobotoBold,
    weight: 'bold',
  },
];

export const FONT_FACES = css`
  ${FONT_FACES_LIST.map(
    ({ name, url, weight }: FontFace) => css`
      @font-face {
        font-family: ${name}, 'sans-serif';
        src: url(${url});
        font-weight: ${weight};
      }
    `
  )}
`;
