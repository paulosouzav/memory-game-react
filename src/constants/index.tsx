export { CARDS } from './cards';
export { COLORS } from './colors';
export { ROTATION } from './rotation';
export { ROUTES } from './routes';
export { BREAKPOINTS, MAX_WIDTHS } from './screen';
export { SHADOWS } from './shadows';
export { SIZES } from './sizes';
