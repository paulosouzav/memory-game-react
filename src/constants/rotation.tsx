import { css } from 'styled-components';

export const ROTATION = {
  half: css`
    transform: rotateY(180deg);
  `,
};
