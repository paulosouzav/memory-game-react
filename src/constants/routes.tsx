import { Game } from '../views';

interface route {
  id: string;
  path: string;
  description: string;
  Component: Function;
}

export const ROUTES: Array<route> = [
  {
    id: 'game',
    path: '/',
    description: 'Game',
    Component: Game,
  },
];
