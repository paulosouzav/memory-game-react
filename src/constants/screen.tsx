interface Indexable {
  [key: string]: number;
}

export const MAX_WIDTHS: Indexable = Object.freeze({
  sm: 540,
  md: 720,
  lg: 960,
  xl: 1140,
});

export const BREAKPOINTS: Indexable = Object.freeze({
  sm: 576,
  md: 758,
  lg: 992,
  xl: 1200,
});
