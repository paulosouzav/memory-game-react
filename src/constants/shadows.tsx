import { css } from 'styled-components';
import { TRANSPARENT_BLACK } from './colors';

export const SHADOWS = Object.freeze({
  none: css`
    box-shadow: none;
  `,
  soft: css`
    box-shadow: 0 0 8px ${TRANSPARENT_BLACK.soft};
  `,
  normal: css`
    box-shadow: 0 0 8px ${TRANSPARENT_BLACK.normal};
  `,
  hard: css`
    box-shadow: 0 0 8px ${TRANSPARENT_BLACK.hard};
  `,
});
