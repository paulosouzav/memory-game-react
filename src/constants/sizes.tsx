export const SIZES = Object.freeze({
  navHeight: '60px',
});
