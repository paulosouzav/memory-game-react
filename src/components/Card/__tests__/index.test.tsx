import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { Card } from '../';
import { DEFAULT_CARD_IMAGE } from '../../../constants/cards';
import Asteroid from '../../../assets/cards/002-asteroid.png';

const CARD_MOCK = {
  id: 'asteroid',
  image: {
    src: Asteroid,
    alt: 'Asteroid',
  },
  isMatched: false,
};

describe('Card', () => {
  it('should render without failing', () => {
    const wrapper = mount(<Card {...CARD_MOCK} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render front and back face of card', () => {
    const expectedFrontFaceProps = {
      ...CARD_MOCK.image,
      matched: CARD_MOCK.isMatched,
      front: true,
    };

    const wrapper = mount(<Card {...CARD_MOCK} />);
    const frontFace = wrapper.find('Face').at(0);
    const backFace = wrapper.find('Face').at(1);

    expect(frontFace.props()).toEqual(expectedFrontFaceProps);
    expect(backFace.props()).toEqual(DEFAULT_CARD_IMAGE);
  });
});
