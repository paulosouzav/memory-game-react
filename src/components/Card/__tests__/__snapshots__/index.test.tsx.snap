// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render without failing 1`] = `
.c1 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
}

.c2 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
}

.c0 {
  width: 155px;
  height: 210px;
  margin: 10px;
  position: relative;
  border-radius: 8px;
  -webkit-transition: -webkit-transform .4s,opacity .4s;
  -webkit-transition: transform .4s,opacity .4s;
  transition: transform .4s,opacity .4s;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  box-shadow: 0 0 8px rgba(0,0,0,0.5);
}

.c0:active {
  -webkit-transform: scale(0.97);
  -ms-transform: scale(0.97);
  transform: scale(0.97);
}

@media (max-width:576px) {
  .c0 {
    width: 90px;
    height: 130px;
  }
}

<Card
  id="asteroid"
  image={
    Object {
      "alt": "Asteroid",
      "src": "002-asteroid.png",
    }
  }
  isMatched={false}
>
  <BaseCard
    id="asteroid"
    isMatched={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "Face": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c2",
              "rules": Array [
                "
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ",
                [Function],
                "
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "Face",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 155px;
  height: 210px;
  margin: 10px;
  position: relative;
  border-radius: 8px;

  ",
              "transition: transform .4s, opacity .4s;",
              "
  transform: scale(1);
  transform-style: preserve-3d;

  ",
              "
    box-shadow: 0 0 8px ",
              "rgba(0,0,0,0.5)",
              ";
  ",
              "
  
  ",
              "
      @media (",
              "max",
              "-width: ",
              "576",
              "px) {
        ",
              "
    width: 90px;
    height: 130px;
  ",
              ";
      }
    ",
              "

  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "BaseCard",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="asteroid"
      isMatched={false}
    >
      <div
        className="c0"
        id="asteroid"
      >
        <Face
          alt="Asteroid"
          front={true}
          matched={false}
          src="002-asteroid.png"
        >
          <StyledComponent
            alt="Asteroid"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ",
                    [Function],
                    "
  ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "Face",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            front={true}
            matched={false}
            src="002-asteroid.png"
          >
            <Component
              alt="Asteroid"
              className="c1"
              front={true}
              matched={false}
              src="002-asteroid.png"
            >
              <Image
                alt="Asteroid"
                className="c1"
                src="002-asteroid.png"
              >
                <img
                  alt="Asteroid"
                  className="c1"
                  src="002-asteroid.png"
                />
              </Image>
            </Component>
          </StyledComponent>
        </Face>
        <Face
          alt="Default"
          src="default-card.png"
        >
          <StyledComponent
            alt="Default"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ",
                    [Function],
                    "
  ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "Face",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            src="default-card.png"
          >
            <Component
              alt="Default"
              className="c2"
              src="default-card.png"
            >
              <Image
                alt="Default"
                className="c2"
                src="default-card.png"
              >
                <img
                  alt="Default"
                  className="c2"
                  src="default-card.png"
                />
              </Image>
            </Component>
          </StyledComponent>
        </Face>
      </div>
    </StyledComponent>
  </BaseCard>
</Card>
`;
